### ISSUE TITLE
Investigate email sends from prober tests in AWS staging

### ISSUE THREAD

**Issue by rockycodes** [civiform/civiform#4768]
In AWS staging we are limited to 200 email sends per day. We are currently hitting that limit because of prober tests that run when staging is deployed which send real emails. This means that we can't send adhoc test emails from staging and sometimes are blocked from doing things like adding a verified email address (which requires sending a verification email).

One option is to request production access for SES (Amazon Simple Email Service) which would up our quota to 50k sends per day.

To avoid extra charges, however, it would be great to leave it in Sandbox mode and figure out a way to run the tests without sending real emails. Some options for doing this might include:
- Updating tests to use localstack
- Adding a flag in tests to prevent emails from actually being sent
- Using a non-verified email address in tests (AWS will only send from a verified email address)

**Comment by gwendolyngoetz** [comment]
@rockycodes Sorry!

**Comment by rockycodes** [comment]
@gwendolyngoetz not your fault! I should have thought through what the impact might be to our staging env.

**Comment by rockycodes** [comment]
Update -- I requested an upgrade to "production" in AWS which will bump our limit of email sends to 50k per day (up to 62k email sends per month is still free: https://aws.amazon.com/ses/pricing/). I'm hoping this will solve the issue for now, but if need be we can also handle staging email sends differently [here](https://github.com/civiform/civiform/blob/main/server/app/services/applicant/ApplicantService.java#L588).

Looking into this raised the issue that we need better error handling around email sends (currently we just get the generic "Oops, an error occurred" page. I created an issue for this (and other issues with error handling we might find): https://github.com/civiform/civiform/issues/4783

Once approved for production access, I will set up an alert so we are notified if we are getting close to the 62k/month free email threshold.

Also, @gwendolyngoetz I confirmed that adding the `civiform-dev@exygy.com` email address to the repo is not what caused this! Adding it in the repo just controls the email address that shows up on the page and who gets emailed if there's an issue. This rate limiting issue is caused by setting `civiform-dev@exygy.com` as the email in our staging config.
